api/v1/model_handlers.go

```diff
@@ -0,0 +1,304 @@
+		Param(ws.QueryParameter("start", "Start time for requested metrics").DataType("string")).
+		Writes(MetricResult{}))
+
+	// The /namespace/{}/pod endpoint exposes aggregated metrics for the Pod entity of the model.
+	ws.Route(ws.GET("/namespace/{namespace-name}/pod/{pod-name}/{metric-name}").
+		To(a.podMetrics).
+		Filter(compressionFilter).
+		Doc("export all aggregated pod-level metrics").
+		Operation("podMetrics").
+		Param(ws.PathParameter("namespace-name", "The name of the namespace to use").DataType("string")).
+		Param(ws.PathParameter("pod-name", "The name of the pod to use").DataType("string")).
+		Param(ws.PathParameter("metric-name", "The name of the requested metric").DataType("string")).
+		Param(ws.QueryParameter("start", "Start time for requested metrics").DataType("string")).
+		Writes(MetricResult{}))
+
+	// The /namespace/{}/pod/{}/container endpoint exposes metrics for the Pod Container entity of the model.
+	ws.Route(ws.GET("/namespace/{namespace-name}/pod/{pod-name}/container/{container-name}/{metric-name}").
+		To(a.podContainerMetrics).
+		Filter(compressionFilter).
+		Doc("export a container-level metric for a pod container").
+		Operation("podContainerMetrics").
+		Param(ws.PathParameter("namespace-name", "The name of the namespace to use").DataType("string")).
+		Param(ws.PathParameter("pod-name", "The name of the pod to use").DataType("string")).
+		Param(ws.PathParameter("metric-name", "The name of the requested metric").DataType("string")).
+		Param(ws.QueryParameter("start", "Start time for requested metrics").DataType("string")).
+		Writes(MetricResult{}))
+
+	// The /node/{}/container endpoint exposes metrics for the Free Container entity of the model.
+	ws.Route(ws.GET("/node/{node-name}/container/{container-name}/{metric-name}").
+		To(a.freeContainerMetrics).
+		Filter(compressionFilter).
+		Doc("export a container-level metric for a free container").
+		Operation("freeContainerMetrics").
+		Param(ws.PathParameter("node-name", "The name of the node to use").DataType("string")).
+		Param(ws.PathParameter("container-name", "The name of the container to use").DataType("string")).
+		Param(ws.PathParameter("metric-name", "The name of the requested metric").DataType("string")).
+		Param(ws.QueryParameter("start", "Start time for requested metrics").DataType("string")).
+		Writes(MetricResult{}))
+
+	container.Add(ws)
+}
+
+// availableMetrics returns a list of available metric names, to be reported to the /metrics endpoint.
+// These metric names can be used to extract metrics from the various model entities.
+func (a *Api) availableMetrics(request *restful.Request, response *restful.Response) {
+	cluster := a.manager.GetCluster()
+	result := cluster.GetAvailableMetrics()
+	response.WriteEntity(result)
+}
+
+// clusterMetrics returns a metric timeseries for a metric of the Cluster entity.
+func (a *Api) clusterMetrics(request *restful.Request, response *restful.Response) {
+	var err error
```

Comment:
right! The var definition was left over before I ported some code to parseRequestStartParam